drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -101,6 +110,50 @@ static int adc_smartbond_channel_setup(const struct device *dev,
 	return 0;
 }
 
+static inline void gpadc_smartbond_pm_policy_state_lock_get(const struct device *dev,
+					      struct adc_smartbond_data *data)
+{
+#if defined(CONFIG_PM_DEVICE_RUNTIME)
+	if (!atomic_test_and_set_bit(data->pm_policy_state_flag, 0)) {
+		/*
+		 * Prevent the SoC from entering the normal sleep state.
+		 */
+		pm_device_runtime_get(dev);
+	}
+
+#elif defined(CONFIG_PM_DEVICE)
```

Comment:
PM constraints should be used even in device runtime mode as the system is not allowed to enter the sleep state until all conversions are exercised. The only difference is that `gpadc_smartbond_pm_action` will be invoked explicitly by the driver by calling `pm_device_runtime_get/put`.

Reply:
Changed the elif into if